fdb-relational-grpc/src/test/java/com/apple/foundationdb/relational/jdbc/ResultSetContinuationTest.java

```diff
@@ -0,0 +1,134 @@
+                Arguments.of(MockContinuation.END),
+                Arguments.of(new MockContinuation(Continuation.Reason.TRANSACTION_LIMIT_REACHED, new byte[]{1}, false, false)),
+                Arguments.of(new MockContinuation(Continuation.Reason.CURSOR_AFTER_LAST, new byte[]{1}, false, true)),
+                Arguments.of(new MockContinuation(Continuation.Reason.QUERY_EXECUTION_LIMIT_REACHED, new byte[0], false, true)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("continuationSource")
+    void testResultSetWithContinuation(Continuation continuation) throws Exception {
+        RelationalResultSet resultSet = TestUtils.resultSet(
+                "TestType",
+                List.of(Types.INTEGER, Types.INTEGER, Types.INTEGER),
+                continuation,
+                TestUtils.row(1, 2, 3), TestUtils.row(4, 5, 6));
+        ResultSet rsProto = TypeConversion.toProtobuf(resultSet);
+
+        try (RelationalResultSetFacade deserializedResultSet = new RelationalResultSetFacade(rsProto)) {
+            List<List<Integer>> rows = toRows(deserializedResultSet, 3);
+            Assertions.assertEquals(List.of(List.of(1, 2, 3), List.of(4, 5, 6)), rows);
+            Continuation convertedContinuation = deserializedResultSet.getContinuation();
+            assertContinuation(continuation, convertedContinuation);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("continuationSource")
+    void testContinuationRoundTrip(Continuation continuation) throws Exception {
+        RelationalResultSet resultSet = TestUtils.resultSet(
+                "TestType",
+                List.of(Types.INTEGER, Types.INTEGER, Types.INTEGER),
+                continuation,
+                TestUtils.row(1, 2, 3), TestUtils.row(4, 5, 6));
+        ResultSet rsProto = TypeConversion.toProtobuf(resultSet);
+
+        try (RelationalResultSetFacade deserializedResultSet = new RelationalResultSetFacade(rsProto)) {
+            // Spend all the rows so we can get the continuation
+            List<List<Integer>> rows = toRows(deserializedResultSet, 3);
+            byte[] returnedContinuation = deserializedResultSet.getContinuation().serialize();
+            byte[] expectedContinuation = continuation.serialize();
+            Assertions.assertArrayEquals(expectedContinuation, returnedContinuation);
+        }
+    }
+
+    @Test
+    void testMidContinuationFails() throws Exception {
+        RelationalResultSet resultSet = TestUtils.resultSet(
+                "TestType",
+                List.of(Types.INTEGER, Types.INTEGER, Types.INTEGER),
+                MockContinuation.BEGIN,
+                TestUtils.row(1, 2, 3), TestUtils.row(4, 5, 6));
+        ResultSet rsProto = TypeConversion.toProtobuf(resultSet);
+
+        try (RelationalResultSetFacade deserializedResultSet = new RelationalResultSetFacade(rsProto)) {
+            Assertions.assertThrows(SQLException.class, () -> deserializedResultSet.getContinuation());
```

Comment:
This is only testing that it fails before getting any results. It seems like that is a good thing to test, but also that we should test after getting one.

Reply:
Done.

Reply:
Yes, that's a good enhancement. Done. 
This was also used to cover the off-by-one bug in the result set facade that only allowed continuations after the last row.